src/librustc/diagnostics.rs

```diff
@@ -227,6 +298,16 @@ match Some(5) {
 }
 
 See also https://github.com/rust-lang/rust/issues/14587
+"##,
+
+E0306: r##"
+In an array literal `[x; N]`, `N` is the number of elements in the array. For
+obvious reasons, this number cannot be negative.
```

Comment:
It may be worth toning down the wording here to just say "This number cannot be negative"